### PR TITLE
SimG4CMS packages: Fix clang warnings about absolute value use.

### PR DESCRIPTION
**Pull request by gartung**
Replace abs/fabs with std::abs which has a signature for ints and floats.
Fixed misplaced parens which result in absolute values of a bool.
The absolute value of unsigned vars is meaningless. Cast to signed vars
to prevent difference from wrapping to unsigned var max value.

### CODE REVIEW

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

File: SimG4CMS/ShowerLibraryProducer/src/CastorShowerLibraryMaker.cc [hunk 1]
```diff
@@ -59,7 +59,7 @@ CastorShowerLibraryMaker::CastorShowerLibraryMaker(const edm::ParameterSet &p) :
 //
   NPGParticle               = PGParticleIDs.size(); 
   for(unsigned int i=0;i<PGParticleIDs.size();i++) {
-     switch (int(fabs(PGParticleIDs.at(i)))) {
+     switch (int(std::abs(PGParticleIDs.at(i)))) {
```
Comment (davidlange6) [inline SimG4CMS/ShowerLibraryProducer/src/CastorShowerLibraryMaker.cc]: Hi @gartung - we can drop the int() now?

**Comment by gartung** [comment]
Done.

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

**Comment by davidlange6** [comment]
+1

**Pull request merged**